src/main/rascal/refactor/Rename.rsc

```diff
@@ -282,18 +289,20 @@ default set[Define] getCursorDefinitions(list[Tree] cursor, Tree(loc) _, TModel(
     return {};
 }
 
-default tuple[set[loc] defFiles, set[loc] useFiles] findOccurrenceFiles(set[Define] cursorDefs, list[Tree] cursor, Tree(loc) _, Renamer r) {
+default tuple[set[loc] defFiles, set[loc] useFiles, set[loc] newNameFiles] findOccurrenceFiles(set[Define] cursorDefs, list[Tree] cursor, str newName, Tree(loc) _, Renamer r) {
     loc f = cursor[0].src.top;
     if (any(d <- cursorDefs, f != d.defined.top)) {
         r.error(cursor[0].src, "Rename not implemented for cross-file definitions. Please overload `findOccurrenceFiles`.");
-        return <{}, {}>;
+        return <{}, {}, {}>;
     }
 
-    return <{f}, {f}>;
+    return <{f}, {f}, {f}>;
 }
 
 default set[Define] findAdditionalDefinitions(set[Define] cursorDefs, Tree tr, TModel tm) = {};
 
+default void validateNewNameOccurrences(set[Define] cursorDefs, str newName, Tree _, Renamer r) {}
```

Comment:
I have the feeling there is a more correct default for a language that doesn't have overloading (or shadowing)?

Reply:
Or, of course, retrieve the TModel here and look for definitions of `newName`

Reply:
Nice 👍🏼